vault/resource_ldap_secret_backend_static_role.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+func ldapSecretBackendStaticRoleResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		consts.FieldPath: {
+			Type:         schema.TypeString,
+			Default:      consts.MountTypeLDAP,
+			Optional:     true,
+			Description:  "The path where the LDAP secrets backend is mounted.",
+			ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+		},
+		consts.FieldRole: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the role.",
+			ForceNew:    true,
+		},
+		consts.FieldUsername: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The username of the existing LDAP entry to manage password rotation for.",
+			ForceNew:    true,
+		},
+		consts.FieldDN: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Distinguished name (DN) of the existing LDAP entry to manage password rotation for.",
+		},
+		consts.FieldRotationPeriod: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "How often Vault should rotate the password of the user entry.",
+		},
+	}
+	return provider.MustAddMountMigrationSchema(&schema.Resource{
```

Comment:
Similar to comment below, this is only needed for resources that handle mounting/unmounting of secret/auth engines. Can be removed from the static role resource

```suggestion
```